pkg/sql/pg_catalog.go

```diff
@@ -1254,3 +1303,43 @@ func (h oidHasher) BuiltinOid(name string, builtin *parser.Builtin) *parser.DInt
 	h.writeStr(builtin.Types.String())
 	return h.getOid()
 }
+
+func (h oidHasher) UserOid(username string) *parser.DInt {
+	h.writeTypeTag(userTypeTag)
+	h.writeStr(username)
+	return h.getOid()
+}
+
+func forEachUser(p *planner, fn func(username string) error) error {
+	query := `SELECT username FROM system.users`
+	plan, err := p.query(query)
+	if err != nil {
+		return nil
+	}
+	defer plan.Close()
+	if err := plan.Start(); err != nil {
+		return err
+	}
+
+	// TODO(cuongdo/asubiotto): Get rid of root user special-casing if/when a row
```

Comment:
I do not understand why you need the special case here. Do apps that use pg_role need to observe the existence of "root"?